engine/src/main/java/org/terasology/rendering/logic/FloatingTextRenderer.java

```diff
@@ -99,18 +100,24 @@ private void render(Iterable<EntityRef> floatingTextEntities) {
 
             FloatingTextComponent floatingText = entity.getComponent(FloatingTextComponent.class);
 
-            String text = floatingText.text;
+            List<String> text = Arrays.asList(floatingText.text.split("\n"));
```

Comment:
Can you check if you can simply write here:

`String[] text = floatingText.text.split("\n");`